cinema/views.py

```diff
@@ -56,3 +89,61 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self) -> QuerySet:
+        queryset = self.queryset
+
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+
+        if movie:
+            queryset = queryset.filter(movie=movie)
+
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if self.action in ("list", "retrieve"):
+
+            queryset = (
+                queryset
+                .select_related("movie", "cinema_hall")
```

Comment:
If you don't use "movie" in your query, there's no need to include it in the select_related() method call.

Reply:
good point, but `movie` is actually is being used there:)